server/service/src/processors/mod.rs

```diff
@@ -106,11 +120,35 @@ impl ProcessorsTrigger {
         }
     }
 
+    /// Waits till all current events in the processor queue are handled.
+    /// Its guaranteed that all queued processor events that where in the queue before calling
+    /// this method are handled when this method returns.
+    /// However, new events might have been added while this method was running.
+    pub async fn await_events_processed(&self) {
```

Comment:
Is not `pub(crate)` because its currently only used in tests and would otherwise give a warning...